### PR TITLE
remove VDR memstore

### PR DESCRIPTION
**Pull request by woutslakhorst**
It was only used in test and had custom logic in it. Switching to the BBolt one (also in memory) actually made some tests fail!

Part of #1709

### CODE REVIEW

File: vdr/store/store.go [hunk 1]
```diff
@@ -288,6 +294,10 @@ func (s *store) Resolve(id did.DID, metadata *vdr.ResolveMetadata) (returnDocume
 				return err
 			}
 
+			if metadataRecord.Deactivated && metadata == nil {
```
Comment (reinkrul) [inline vdr/store/store.go:297]: this belongs in a different PR

File: vdr/store/store.go [hunk 2]
```diff
@@ -47,20 +47,26 @@ const (
...
-	return &store{storeProvider: storeProvider}
+func NewStore(storeProvider storage.Provider, inMemory bool) vdr.Store {
+	return &store{storeProvider: storeProvider, inMemory: inMemory}
 }
 
 func (s *store) Configure(_ core.ServerConfig) error {
 	var err error
-	s.db, err = s.storeProvider.GetKVStore(didStoreName, storage.PersistentStorageClass)
+	if s.inMemory {
+		s.db, err = s.storeProvider.GetKVStore(didStoreName, storage.VolatileStorageClass)
```
Comment (reinkrul) [inline vdr/store/store.go:66]: Volatile doesn't mean in-memory per se. Just that the storage may be lost without losing data (e.g. connection backoffs).

Other tests use `StaticKVProvider` with a precreated test-BBolt store (`CreateTestBBoltStore`, `storage/test.go:38`).

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: vdr/store/store.go (hunk 1 above)
Comment (woutslakhorst) [reply vdr/store/store.go:297]: Not really, since the in memory store did it correctly. With the change to the other store, tests fail.

**Review by reinkrul — APPROVED** [review]
